lib/xlibc/src/threads/thrd_create.c

```diff
@@ -1,13 +1,18 @@
 #include <threads.h>
 #include <_threads.h>
+#include <_tls.h>
 #include "xboxkrnl/xboxkrnl.h"
 
 int thrd_create (thrd_t *thr, thrd_start_t func, void *arg)
 {
     NTSTATUS ntstatus;
     ULONG stacksize;
+    ULONG tlssize;
+    // FIXME: We're directly reading the XBE StackCommit field here.
+    //        A cleaner way with proper structs would be nice.
     stacksize = *((ULONG *)0x00010130);
-    ntstatus = PsCreateSystemThreadEx(thr, 0, stacksize, 0, NULL, (PKSTART_ROUTINE)func, arg, FALSE, FALSE, _xlibc_thread_startup);
+    tlssize = _tls_used.SizeOfZeroFill + _tls_used.EndAddressOfRawData - _tls_used.StartAddressOfRawData + 4;
```

Comment:
Where does the `+ 4` come from? Even if there's no TLS data, shouldn't `(EndAddressOfRawData - StartAddressOfRawData)` be 1 (or larger due to alignment)?
I'd expect `- 1` or something.

Reply:
Right, there's a self-reference. I'll do the reordering and add a comment.